src/creep.prototype.js

```diff
@@ -573,6 +586,122 @@ Creep.prototype.routeTo = function routeCreep(target, opts) {
   });
 };
 
+function getTargetAndRange(creep) {
+  let range = 1;
+  let target;
+  if (creep.memory.target) {
+    if (_.isNumber(creep.memory.target)) {
+      target = Game.getObjectById(creep.memory.target)
+    } else if (Array.isArray(creep.memory.target)) {
+      target = new RoomPosition(...creep.memory.target);
+    }
+    if (_.isNumber(creep.memory.range)) {
+      range = creep.memory.range;
+    }
+  }
+  return [target, range];
+}
+
+function moveCreepOutOfTheWay(creep, target, range, creepsToMove, nextPositions, creepMoves) {
+  const availableSpots = [...walkBox(creep.pos)].filter(([x, y]) => {
+    const newPos = new RoomPosition(x, y, creep.room.name);
+    if (target && newPos.getRangeTo(target) > range) {
+      return false;
+    }
+    for (let curr of newPos.look()) {
+      if (
+        (curr.type === 'terrain' && curr.terrain === 'wall')
+        || (
+          curr.type === 'structure'
+          && !curr.structure instanceof StructureRoad
+          && !curr.structure instanceof StructureContainer
+          && !(curr.structure instanceof StructureRampart && (curr.my || (!curr.my && curr.isPublic)))
```

Comment:
impassable position is any terrain that is a wall, or any structure that is not a road, container, my rampart, or a hostile rampart that is set to public.